### ISSUE TITLE
tonumber not match lua 5.4 behaviour

### ISSUE THREAD

**Issue by aarnaud** [dragonflydb/dragonfly#2664]
**Describe the bug**
`tonumber` in lua 5.4 must never return a `string`, that have to return a `number` or `fail` 

But with a `float` input it return a `string` ! 

Ref: https://www.lua.org/manual/5.4/manual.html#tonumber

**To Reproduce**
```
127.0.0.1:6379> eval 'return tonumber(34343.223121)' 0
"34343.223121"
```

**Expected behavior**
this is from Redis v7

```
127.0.0.1:6379> eval 'return tonumber(34343.223121)' 0
(integer) 34343
```

**Screenshots**

**Environment (please complete the following information):**
 - OS: ubuntu 22.04
 - Kernel: 6.5.0-21-generic
 - Containerized?: Kubernetes
 - Dragonfly Version: v1.14.1

**Comment by romange** [comment]
lua does not return a string but RESP2 [can not return floats](https://redis.io/docs/reference/protocol-spec/#resp-protocol-description) so it returns strings instead. I do not think that Redis7 behavior is the correct one since `tonumber` should not round down numbers. I suggest that we return floats for RESP3 and strings for RESP2 - see https://redis.io/docs/reference/protocol-spec/#doubles 

Will that work for you @aarnaud  ?

**Comment by aarnaud** [comment]
float instead of string when using `myvar = tonumber("1222.1212")`  should work to move from redis to dragonflagdb

**Comment by romange** [comment]
@aarnaud what redis client do you use?

**Comment by romange** [comment]
@aarnaud i just checked - it's working as expected:
```
> redis-cli  -3

127.0.0.1:6379> eval 'return tonumber(34343.223121)' 0
(double) 34343.223121
```

please upgrade to resp3 to benefit from using floats.

**Comment by aarnaud** [comment]
I think i don't have the control of the api version use by the app. I just notice the difference between Redis and Dragonfly because there is crashes in logs. 
I opened this issue because Dragonfly is normally `fully compatible with Redis APIs`

**Comment by romange** [comment]
What's the app and what's the redis client? Most clients should support resp3 so I am curious to learn why the friction exists.

**Comment by aarnaud** [comment]
It's Sentry, they have there own legacy client (RB, redis blaster) that is still used a some point, I checked quickly le config option and I don't find a way to specify the protocol. for example on Snuba.  

For now, the deployment with the helm release redis-ha will do the job to have a resilient redis. But I had big hope on Dragonfly

There is also in lua redis.log that don't exist

**Comment by romange** [comment]
redis.log is already merged and will be present in 1.15.
I will add legacy support for doubles for RESP2 under a flag for 1.5 as well.

**Comment by aarnaud** [comment]
waw, thanks @romange
